src/main/java/com/squareup/kotlinpoet/TypeSpec.kt

```diff
@@ -598,6 +583,21 @@ class TypeSpec private constructor(builder: TypeSpec.Builder) {
     }
 
     fun build(): TypeSpec {
+      if (enumConstants.isNotEmpty()) {
+        check(isEnum) { "${this.name} is not enum and cannot have enum constants" }
+        enumConstants.keys.forEach {
+          require(it.isName) { "not a valid enum constant: $name" }
+        }
+      }
+
+      if (superclassConstructorParameters.isNotEmpty()) {
+        ensureCanHaveSuperclass()
```

Comment:
Off topic: we should use the word check, not ensure, right?

Reply:
Agree, let's be consistent

Reply:
840f2cd